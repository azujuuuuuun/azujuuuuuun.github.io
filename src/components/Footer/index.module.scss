.Footer {
  padding-top: 20px;

  &__copyright {
    font-size: 1.2rem;
  }
}
