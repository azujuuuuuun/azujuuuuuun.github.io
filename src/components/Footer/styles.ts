const styles = {
  container: {
    paddingTop: 20,
  },
  linksList: {
    margin: 0,
  },
  linksItem: {
    display: "inline-block",
    marginRight: 16,
  },
};

export default styles;
