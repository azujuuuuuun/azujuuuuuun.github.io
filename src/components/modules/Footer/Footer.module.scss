@use "../../../styles/variables" as *;

.Footer {
  display: flex;
  justify-content: space-between;
  padding: $space-xs $space-l;

  &__copyright {
    font-size: $font-size-xs;
  }

  &__updateDate {
    font-size: $font-size-xs;
  }
}
