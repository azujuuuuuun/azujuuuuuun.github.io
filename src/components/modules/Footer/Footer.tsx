import clsx from "clsx";
import dayjs from "dayjs";
import type React from "react";

interface FooterProps {
  className?: string;
  updateDate: string;
}

export const Footer: React.FC<FooterProps> = ({ className, updateDate }) => {
  return (
    <footer className={clsx("flex justify-between px-l py-xs", className)}>
      <small className="text-xs">Â©jun</small>
      <time
        className="text-xs"
        dateTime={dayjs(updateDate).format("YYYY-MM-DD")}
      >
        Update date: {dayjs(updateDate).format("YYYY/MM/DD")}
      </time>
    </footer>
  );
};
