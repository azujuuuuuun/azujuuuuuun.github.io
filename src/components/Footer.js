import React from 'react'

import Links from '../components/Links'
import styles from './FooterStyles'

export default props => {
  const { isHome } = props
  return (
    <footer>
      {!isHome && (
        <Links listStyle={styles.linksList} itemStyle={styles.linksItem} />
      )}
      <small>Â©Copyright jun. All rights reserved.</small>
    </footer>
  )
}
