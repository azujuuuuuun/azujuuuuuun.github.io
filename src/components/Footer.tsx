import React from 'react'

import Links from '../components/Links'
import styles from './FooterStyles'

interface Props {
  isHome?: boolean
}

export default (props: Props) => {
  const { isHome } = props
  return (
    <footer style={styles.container}>
      {!isHome && (
        <Links
          listStyle={styles.linksList}
          isFooter
          itemStyle={styles.linksItem}
        />
      )}
      <small>Â©Copyright jun. All rights reserved.</small>
    </footer>
  )
}
