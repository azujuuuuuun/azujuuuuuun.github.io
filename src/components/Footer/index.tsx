import React from 'react';

import Links from '../../components/Links';
import styles from './styles';

interface Props {
  isHome?: boolean;
}

const Footer: React.FC<Props> = props => {
  const { isHome } = props;
  return (
    <footer style={styles.container}>
      {!isHome && (
        <Links
          listStyle={styles.linksList}
          isFooter
          itemStyle={styles.linksItem}
        />
      )}
      <small>Â©Copyright jun. All rights reserved.</small>
    </footer>
  );
};

export default Footer;
