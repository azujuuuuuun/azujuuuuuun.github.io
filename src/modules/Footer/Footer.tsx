import React from "react";
import clsx from "clsx";
import dayjs from "dayjs";
import styles from "./Footer.module.scss";

interface FooterProps {
  className?: string;
  updateDate: string;
}

export const Footer: React.FC<FooterProps> = ({ className, updateDate }) => {
  return (
    <footer className={clsx(styles.Footer, className)}>
      <small className={styles.Footer__copyright}>Â©jun</small>
      <time
        className={styles.Footer__updateDate}
        dateTime={dayjs(updateDate).format("YYYY-MM-DD")}
      >
        Update date: {dayjs(updateDate).format("YYYY/MM/DD")}
      </time>
    </footer>
  );
};
