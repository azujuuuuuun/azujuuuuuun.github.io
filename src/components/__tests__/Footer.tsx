import React from 'react';
import renderer from 'react-test-renderer';

import Footer from '../Footer';

describe('Footer', (): void => {
  it('renders correctly', (): void => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
