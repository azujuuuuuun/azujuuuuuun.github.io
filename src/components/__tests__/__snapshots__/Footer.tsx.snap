// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer
  style={
    Object {
      "paddingTop": 20,
    }
  }
>
  <ul
    style={
      Object {
        "margin": 0,
      }
    }
  >
    <li
      style={
        Object {
          "display": "inline-block",
          "marginRight": 16,
        }
      }
    >
      <a
        aria-current="page"
        className=""
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        style={Object {}}
      >
        ホーム
      </a>
    </li>
    <li
      style={
        Object {
          "display": "inline-block",
          "marginRight": 16,
        }
      }
    >
      <a
        href="/about"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        自己紹介
      </a>
    </li>
    <li
      style={
        Object {
          "display": "inline-block",
          "marginRight": 16,
        }
      }
    >
      <a
        href="/skills"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        スキル
      </a>
    </li>
    <li
      style={
        Object {
          "display": "inline-block",
          "marginRight": 16,
        }
      }
    >
      <a
        href="/favorites"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        好きなもの
      </a>
    </li>
    <li
      style={
        Object {
          "display": "inline-block",
          "marginRight": 16,
        }
      }
    >
      <a
        href="/notes"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        メモ
      </a>
    </li>
    <li
      style={
        Object {
          "display": "inline-block",
          "marginRight": 16,
        }
      }
    >
      <a
        href="/links"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        リンク
      </a>
    </li>
  </ul>
  <small>
    ©Copyright jun. All rights reserved.
  </small>
</footer>
`;

exports[`Footer renders isHome prop correctly 1`] = `
<footer
  style={
    Object {
      "paddingTop": 20,
    }
  }
>
  <small>
    ©Copyright jun. All rights reserved.
  </small>
</footer>
`;
